Assets/Scripts/AI/Cartographer.cs

```diff
@@ -0,0 +1,87 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Object = UnityEngine.Object;
+
+namespace AI
+{
+    public class Cartographer
+    {
+        private readonly List<AiWrapper> _gameObjects = new List<AiWrapper>();
+        private readonly Map _map;
+
+        /// <summary>
+        /// Constructs a cartographer for a given width and height
+        /// </summary>
+        /// <param name="width">Width of the map.</param>
+        /// <param name="height">Height of the map.</param>
+        public Cartographer(int width, int height)
+        {
+            _map = new Map(width, height);
+            FillStaticMap();
+        }
+
+        /// <summary>
+        /// Converts the matrix to a array which is readable by neuronal networks.
+        /// </summary>
+        /// <returns>float matrix [x,y, information]</returns>
+        public float[,] MatrixNnReady()
+        {
+            int infoDimension = 0;
+
+            List<MapEntry> mapEntries = _map.ToList();
+            foreach (MapEntry entry in mapEntries.Where(entry => entry.Dimension() > infoDimension))
+            {
+                infoDimension = entry.Dimension();
+            }
+
+            float[, ] matrix = new float[mapEntries.Count, infoDimension];
+            for (int i = 0; i < mapEntries.Count; i++)
+            {
+                List<float> attributes = mapEntries[i].Attributes;
+                for (int j = 0; j < infoDimension; j++)
+                {
+                    matrix[i, j] = attributes == null || attributes.Count <= j ? 0f : attributes[j];
+                }
+            }
+
+            return matrix;
+        }
+
+        /// <summary>
+        /// Creates a map of all static game objects in the scene.
+        /// </summary>
+        private void FillStaticMap()
+        {
+            _gameObjects.AddRange(collection: Object.FindObjectsOfType<MonoBehaviour>().OfType<AiWrapper>().ToList());
```

Comment:
You can try to do `FindObjectsOfType<AiWrapper>` as AiWrapper is a `MonoBehaviour` and thus a Unity Object.

Reply:
I will try, but I think I already did it and something went wrong, but can't remember.